kinesis-video-gst-demo/kinesis_video_gstreamer_audio_video_sample_app.cpp

```diff
@@ -470,6 +477,14 @@ static GstFlowReturn on_new_sample(GstElement *sink, CustomData *data) {
         if (CHECK_FRAME_FLAG_KEY_FRAME(kinesis_video_flags)) {
             data->key_frame_pts = buffer->pts;
         }
+    } else if (data->use_absolute_fragment_times) {
```

Comment:
I can't quite understand this logic. Could you please add come comments? Seems that we are trying to start absolute cluster times but from 0?

Also, LN 446 - this is an odd if statement.

if (!(dts_sign > 0))

should be

if (dts_sign <= 0)

Reply:
The logic is setting first frame pts as first_pts and producer_start_time as current time when the first frame arrives. Then with each frame do a minus on first_pts to get a delta between it and first frame’s pts and add producer start time to it to form the absolute timestamp. Thought the naming would be enough so I didn’t add the comment, do you see any specific name make it hard to understand or some of condition or assignments make it hard? I can improve those, or if not any, I can add a general comment also.

Reply:
Perhaps I was not clear. The part you've described made sense. The question is why in absolute timestamp mode we need to keep the "presentation" start time and add it to the frame delta from the first frame? If we rely on the media clock then all should just go as is right?

Reply:
There is no guarantee that gstreamer will return us with first pts as 0. Did some search and found someone reporting pts not starting from 0 before, not sure if it is a bug or acceptable behavior of gstreamer. https://gitlab.freedesktop.org/gstreamer/gst-plugins-good/issues/594

Reply:
I guess, the main question to ask is what we really think of as absolute time on the producer device. We could simply rely on whatever the media clock states and timestamp that - after all, this is what's reported by the media pipeline. However, we need to be careful to ensure that all media pipelines are timestamped with the same clock (need to check). Then, we need to ensure the customers understand the behavior when the device is restarted and when a new streaming session happens. For example, if the actual reported clock is the wall clock then regardless of the streaming session we would have proper timeline - assuming the producer clock doesn't drift (some devices would have NTP running which would adjust the clock on boot up). Perhaps, we should simply take a start timestamp from the customer to timestamp against that or if the timestamp is not specified just use whatever the media pipeline reports?

Reply:
This logic is relying on the media pipeline clock assuming it handles time and should be starting from 0 timestamp drift from producer start time(current time). I think it is same as what you mentioned latest.